src/Protocol/ActivityPub/Processor.php

```diff
@@ -525,17 +533,19 @@ private static function fetchParent(array $activity): string
 			}
 		} elseif (self::isActivityGone($activity['reply-to-id'])) {
 			Logger::notice('The activity is gone. We will not spawn a worker. The queue entry will be deleted', ['parent' => $activity['reply-to-id']]);
-			if (!empty($activity['entry-id'])) {
+			if ($in_background) {
+				// fetching in background is done for all activities where we have got the conversation
+				// There we only delete the single activity and not thr whole thread since we can store the
+				// other posts in the thread even with missing posts.
+				Queue::remove($activity);
+			} elseif (!empty($activity['entry-id'])) {
 				Queue::deleteById($activity['entry-id']);
 			}
 			return '';
-		} else {
+		} elseif (!$in_background) {
 			Logger::notice('Recursion level is too high.', ['parent' => $activity['reply-to-id'], 'recursion-depth' => $recursion_depth]);
```

Comment:
Inverting the condition here allows us to get rid of the not operator:
```suggestion
		} elseif ($in_background) {
			Logger::notice('Fetching is done in the background.', ['parent' => $activity['reply-to-id']]);
		} else {
			Logger::notice('Recursion level is too high.', ['parent' => $activity['reply-to-id'], 'recursion-depth' => $recursion_depth]);
```

Reply:
Done